storey/utils.py

```diff
@@ -138,6 +138,12 @@ def get_remaining_path(url):
 
 def url_to_file_system(url, storage_options):
     scheme, remaining_path = get_remaining_path(url)
+    if url.startswith("ds://"):
+        parsed_url = urlparse(url)
+        if parsed_url.password:
+            scheme = parsed_url.password
```

Comment:
Can we add a test to show what this looks like?

Reply:
done